bemserver_ui/internal_api/analysis/weather.py

```diff
@@ -0,0 +1,288 @@
+                if data["forecast"]:
+                    respData[key][data["parameter"]]["forecast"]["timeseries"][
+                        "id"
+                    ] = data["timeseries_id"]
+                    respData[key][data["parameter"]]["forecast"]["timeseries"][
+                        "name"
+                    ] = data["timeseries"]["name"]
+                    respData[key][data["parameter"]]["forecast"]["timeseries"][
+                        "desc"
+                    ] = data["timeseries"]["description"]
+                    respData[key][data["parameter"]]["forecast"]["timeseries"][
+                        "unit"
+                    ] = data["timeseries"]["unit_symbol"]
+                    liste_forecast_ts_id.append(data["timeseries_id"])
+                else:
+                    respData[key][data["parameter"]]["current"]["timeseries"][
+                        "id"
+                    ] = data["timeseries_id"]
+                    respData[key][data["parameter"]]["current"]["timeseries"][
+                        "name"
+                    ] = data["timeseries"]["name"]
+                    respData[key][data["parameter"]]["current"]["timeseries"][
+                        "desc"
+                    ] = data["timeseries"]["description"]
+                    respData[key][data["parameter"]]["current"]["timeseries"][
+                        "unit"
+                    ] = data["timeseries"]["unit_symbol"]
+                    liste_ts_id.append(data["timeseries_id"])
+
+    ts_data_state = flask.g.api_client.timeseries_datastates.getall()
+
+    for data in ts_data_state.data:
+        if data["name"] == "Raw":  # Change to Clean before commit
+            ts_data_state_id = data["id"]
+
+    if len(liste_ts_id) > 0:
+        weather_data = flask.g.api_client.timeseries_data.download_aggregate(
+            start_time=dt_start.isoformat(),
+            end_time=dt_end.isoformat(),
+            data_state=ts_data_state_id,
+            timeseries_ids=[data for data in liste_ts_id if data is not None],
+            bucket_width_unit=bucket_width_unit,
+            bucket_width_value=bucket_width_value,
+            # convert_to = ["degC", "percent"] for Outdoor conditions
+        )
+    else:
+        weather_data = None
+
+    if forecast == "true" and len(liste_forecast_ts_id) > 0:
+        weather_forecast_data = flask.g.api_client.timeseries_data.download_aggregate(
+            start_time=dt_start.isoformat(),
+            end_time=dt_end_forecast.isoformat(),
+            data_state=ts_data_state_id,
+            timeseries_ids=[data for data in liste_forecast_ts_id if data is not None],
+            bucket_width_unit=bucket_width_unit,
+            bucket_width_value=bucket_width_value,
+            # convert_to = ["degC", "percent"] for Outdoor conditions
+        )
+    else:
+        weather_forecast_data = None
+
+    if weather_data is not None:
+        for key in weather_data.data.keys():
```

Comment:
Use whole dict iteration info here.
For example (key and name var names can be changed for better):
`for key, value in weather_data.data.items():`
...
and at the end (line 273), just use `value` instead of `weather_data.data[key]`